src/devices/custom_devices_diy.ts

```diff
@@ -1276,21 +1287,103 @@ const definitions: Definition[] = [
         model: 'LYWSD03MMC',
         vendor: 'Custom devices (DiY)',
         description: 'Xiaomi temperature & humidity sensor with custom firmware',
-        fromZigbee: [fz.temperature, fz.humidity, fz.battery, fz.hvac_user_interface],
-        toZigbee: [tz.thermostat_temperature_display_mode],
-        configure: async (device, coordinatorEndpoint, logger) => {
-            const endpoint = device.getEndpoint(1);
-            const bindClusters = ['msTemperatureMeasurement', 'msRelativeHumidity', 'genPowerCfg'];
-            await reporting.bind(endpoint, coordinatorEndpoint, bindClusters);
-            await reporting.temperature(endpoint, {min: 10, max: 300, change: 10});
-            await reporting.humidity(endpoint, {min: 10, max: 300, change: 50});
-            await reporting.batteryVoltage(endpoint);
-            await reporting.batteryPercentageRemaining(endpoint);
-        },
-        exposes: [
-            e.temperature(), e.humidity(), e.battery(),
-            e.enum('temperature_display_mode', ea.ALL, ['celsius', 'fahrenheit'])
-                .withDescription('The temperature format displayed on the screen'),
+        extend: [
+            quirkAddEndpointCluster({
+                endpointID: 1,
+                outputClusters: [
+                    'hvacUserInterfaceCfg',
+                ],
+                inputClusters: [
+                    'genPowerCfg',
+                    'msTemperatureMeasurement',
+                    'msRelativeHumidity',
+                    'hvacUserInterfaceCfg',
+                ],
+
+            }),
+            batteryPercentage(),
+            batteryVoltage(),
```

Comment:
When `batteryPercentage` is supported, I don't think we should also expose `batteryVoltage()`.